src/deno_dir.rs

```diff
@@ -83,36 +83,54 @@ impl DenoDir {
     self: &DenoDir,
     filename: &str,
     source_code: &str,
-  ) -> PathBuf {
+  ) -> (PathBuf, PathBuf) {
     let cache_key = source_code_hash(filename, source_code);
-    self.gen.join(cache_key + ".js")
+    (
+      self.gen.join(cache_key.to_string() + ".js"),
+      self.gen.join(cache_key.to_string() + ".js.map"),
+    )
   }
 
   fn load_cache(
     self: &DenoDir,
     filename: &str,
     source_code: &str,
-  ) -> std::io::Result<String> {
-    let path = self.cache_path(filename, source_code);
-    debug!("load_cache {}", path.display());
-    fs::read_to_string(&path)
+  ) -> (std::io::Result<String>, std::io::Result<String>) {
+    let (output_code, source_map) = self.cache_path(filename, source_code);
+    debug!(
+      "load_cache code: {} map: {}",
+      output_code.display(),
+      source_map.display()
+    );
+    (
+      fs::read_to_string(&output_code),
+      fs::read_to_string(&source_map),
+    )
   }
 
   pub fn code_cache(
     self: &DenoDir,
     filename: &str,
     source_code: &str,
     output_code: &str,
+    source_map: &str,
   ) -> std::io::Result<()> {
-    let cache_path = self.cache_path(filename, source_code);
+    let (cache_path, source_map_path) = self.cache_path(filename, source_code);
     // TODO(ry) This is a race condition w.r.t to exists() -- probably should
     // create the file in exclusive mode. A worry is what might happen is there
     // are two processes and one reads the cache file while the other is in the
     // midst of writing it.
-    if cache_path.exists() {
+    if cache_path.exists() & source_map_path.exists() {
       Ok(())
     } else {
-      fs::write(cache_path, output_code.as_bytes())
+      let result = (
+        fs::write(cache_path, output_code.as_bytes()),
+        fs::write(source_map_path, source_map.as_bytes()),
```

Comment:
I think this refactors to:
```
fs::write(cache_path, output_code.as_bytes()?
fs::write(source_map_path, source_map.as_bytes()?
Ok(())
```
Similarly `load_cache` could return `Result<String, String>` (or DenoResult) using ? pattern.

Reply:
Only thing I could get to work:

```rust
fs::write(cache_path, output_code.as_bytes())?;
fs::write(source_map_path, source_map.as_bytes())?;
Ok(())
```

Does that have the same semantic meaning?

Reply:
yep, each ? returns early (with the Err).

Reply:
I couldn't use `DenoResult` in `load_cache` because we would be returning a `std::io::Error` not a `DenoError`, but otherwise the other suggestion worked.